selenium_tests/UserDriverTest.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+import contextlib
+from selenium_tests.RemoteAppDriverTest import RemoteAppDriverTest
+from selenium.webdriver.common.by import By
+
+
+class UserDriverTest(RemoteAppDriverTest):
+    def select_application(self, index=0):
+        self.click_element_located(By.ID, "application-entry-{}".format(index))
+
+    def open_application_settings(self):
+        self.click_element_located(By.ID, "application-settings")
+
+    def stop_application(self):
+        self.click_element_located(By.ID, "stop-button")
+
+    def start_application(self):
+        self.click_element_located(By.ID, "start-button")
+
+    def wait_until_application_running(self):
+        self.wait_until_element_present(By.ID, "application")
+
+    def wait_until_application_stopped(self):
+        self.wait_until_text_inside(By.ID, "start-button", "Start")
+
+    def wait_until_application_list_loaded(self):
+        self.wait_until_element_invisible(By.ID, "loading-spinner")
+
+    @contextlib.contextmanager
```

Comment:
you don't need the context manager anymore. You can simply use the setup/teardown to login and logout specifically for your test cases.